### DISCUSSION TITLE
How come block names can't be longer than 32 characters?

### DISCUSSION THREAD

**Discussion by amna-ukaea** [libMesh/libmesh#3505] [Q&A]
Hi,
The block names I'm using are longer than 32 characters and Libmesh seems to have a max character allowance of 32 characters and so doesn't read the full names. Is there a way to fix this?

**Comment by jwpeterson** [comment]
This limit is part of the Exodus (and ulitmately the NetCDF) library itself. In Exodus, the relevant define is `MAX_STR_LENGTH` defined in include/exodusII.h:
```
#define MAX_STR_LENGTH 32L
```
So, one thing you could try is increasing this value and then recompiling everything. This might just "work" or it might lead to some other issues, I haven't tried it before.
